ivy_tests/test_ivy/helpers/function_testing.py

```diff
@@ -872,6 +896,31 @@ def test_frontend_function(
                 ):
                     assert ret.ivy_array.data is out.ivy_array.data
                 assert ret is out
+        elif test_flags.with_copy:
+            assert not isinstance(ret, tuple)
+
+            if test_flags.generate_frontend_arrays:
+                assert _is_frontend_array(ret)
+            else:
+                assert ivy_backend.is_array(ret)
+
+            if test_flags.generate_frontend_arrays:
+                array_fn = _is_frontend_array
+            else:
+                array_fn = ivy_backend.is_array
+            if "copy" in list(inspect.signature(frontend_fn).parameters.keys()):
+                copy_kwargs["copy"] = True
+            first_array = ivy_backend.func_wrapper._get_first_array(
+                *copy_args, array_fn=array_fn, **copy_kwargs
+            )
+            ret_ = get_frontend_ret(
+                backend_to_test, frontend_fn, *copy_args, **copy_kwargs
+            )
+            if _is_frontend_array(first_array):
+                first_array = first_array.ivy_array
+            if _is_frontend_array(ret_):
```

Comment:
This `if` is not necessary either, since we already asserted that the return is a frontend array.